src/services/event/event-router.ts

```diff
@@ -320,17 +320,25 @@ eventsRouter.get("/:EVENTID/", weakJwtVerification, async (req: Request, res: Re
 eventsRouter.get("/", weakJwtVerification, async (_: Request, res: Response) => {
     const payload: JwtPayload = res.locals.payload as JwtPayload;
 
-    // Get collection from the database, and return it as an array
-    const publicEvents: Event[] = await Models.Event.find({ isStaff: false });
-
-    if (hasStaffPerms(payload)) {
-        return res.status(StatusCode.SuccessOK).send({ events: publicEvents });
+    let publicEvents: Event[];
+    if (payload === undefined) {
+        console.log("no jwt token");
+        publicEvents = await Models.Event.find({ isStaff: false, isPrivate: false });
+    } else if (hasStaffPerms(payload)) {
+        console.log("has staff perms");
+        publicEvents = await Models.Event.find();
+    } else if (isPro(payload)) {
+        publicEvents = await Models.Event.find({ isStaff: false, isPrivate: false });
+    } else {
+        // anyone else (not pro)
+        publicEvents = await Models.Event.find({
+            isStaff: false,
+            isPrivate: false,
+            $or: [{ isPro: false }, { isPro: { $exists: false } }],
```

Comment:
Remove pro exists false check

Reply:
there needs to be an exists false check because isPro isn't always in the model - isPro exists is needed because if it doesn't exist then you can assume it's false, and thus regular users can see the event

Reply:
make isPro required then, i think